modules/programs/vscode.nix

```diff
@@ -124,26 +133,35 @@ in {
   config = mkIf cfg.enable {
     home.packages = [ cfg.package ];
 
-    # Adapted from https://discourse.nixos.org/t/vscode-extensions-setup/1801/2
-    home.file = let
-      subDir = "share/vscode/extensions";
-      toPaths = ext:
-        # Links every dir in ext to the extension path.
-        map (k: { "${extensionPath}/${k}".source = "${ext}/${subDir}/${k}"; })
-        (if ext ? vscodeExtUniqueId then
-          [ ext.vscodeExtUniqueId ]
-        else
-          builtins.attrNames (builtins.readDir (ext + "/${subDir}")));
```

Comment:
This logic was removed and should be added back to `addSymlinkToExtension`.

Reply:
If I understand correctly, this logic isn't necessary anymore as it was a short circuit when dealing with individual extensions. However because we now combine all the extensions together with `buildEnv` like `nixpkgs`, we can just iterate through the `share/vscode/extensions` folder.

Reply:
Can this then be marked as outdated?

Reply:
The short-circuiting was important to avoid IFD, now the issue is back: https://github.com/nix-community/home-manager/issues/3094